### PR TITLE
[map] few updates

### PR DESCRIPTION
**Pull request by chejennifer**
- when zooming and panning, disable hover action
- make tooltip stay in map boundary
- show alert when user selects a place we don't have maps for [issue 977](https://github.com/datacommonsorg/website/issues/977)
- on map click, open the page in the same window [issue 978](https://github.com/datacommonsorg/website/issues/978)
- only push hash to history if it's new. With this change, user only has to click back once to get to the last state

dev updated (http://dev.datacommons.org/tools/map#%26sv%3DCumulativeCount_MedicalConditionIncident_COVID_19_PatientDeceased%26pc%3D1)

### CODE REVIEW

File: static/js/chart/draw_choropleth.ts [hunk 1]
```diff
@@ -97,7 +97,7 @@ function drawChoropleth(
   unit: string,
   statVar: string,
   canClick: boolean,
-  getRedirectLink: (geoDcid: GeoJsonFeatureProperties) => string,
+  redirectAction: (geoDcid: GeoJsonFeatureProperties) => void,
```
Comment (beets) [inline static/js/chart/draw_choropleth.ts:100]: update the comment above as well?

File: static/js/tools/map/app.tsx [hunk 2]
```diff
@@ -100,7 +100,9 @@ function applyHash(context: ContextType): void {
 function updateHash(context: ContextType): void {
   let hash = updateHashStatVarInfo("", context.statVarInfo.value);
   hash = updateHashPlaceInfo(hash, context.placeInfo.value);
-  if (hash) {
-    history.pushState({}, "", `/tools/map#${encodeURIComponent(hash)}`);
+  const newHash = encodeURIComponent(hash);
+  const currentHash = location.hash.replace("#", "");
+  if (newHash && newHash !== currentHash) {
+    history.pushState({}, "", `/tools/map#${newHash}`);
```
Comment (beets) [inline static/js/tools/map/app.tsx:106]: cool :) though, consider adding the URI path as a constant.

File: static/js/tools/map/chart.tsx [hunk 3]
```diff
@@ -222,11 +224,13 @@ const getTooltipHtml = (
     hasValue = true;
   }
   if (!hasValue || !(place.dcid in metadataMapping)) {
-    return titleHtml + `${statVarInfo.name}: ${value}<br />`;
+    return titleHtml + `${statVarInfo.name}: <wbr>${value}<br />`;
```
Comment (beets) [inline static/js/tools/map/chart.tsx:227]: this is cool too!

**Review by beets — COMMENTED** [review]
very cool :)

File: static/js/chart/draw_choropleth.ts (hunk 1 above)
Comment (chejennifer) [reply static/js/chart/draw_choropleth.ts:99]: done

File: static/js/tools/map/app.tsx (hunk 2 above)
Comment (chejennifer) [reply static/js/tools/map/app.tsx]: done

File: static/js/tools/map/chart.tsx (hunk 3 above)
Comment (chejennifer) [reply static/js/tools/map/chart.tsx:227]: thanks!

**Review by beets — APPROVED** [review]

**Pull request merged**